pkg/karmadactl/config/list.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/kubernetes"
+	configInit "github.com/karmada-io/karmada/pkg/karmadactl/config/init"
+)
+
+var (
+	listExample = templates.Examples(`
+	# List Karmada all required images from the specified private image registry
+	%[1]s list --private-image-registry=myregistry.com`)
+)
+
+// NewCmdConfigImagesList creates a new list command
+func NewCmdConfigImagesList(parentCommand string) *cobra.Command {
+	opts := configInit.CommandConfigImageOption{
+		InitOption: kubernetes.NewDefaultCommandInitOption(),
```

Comment:
Could we consider removing the function `NewDefaultCommandInitOption`, as default values can all be set through `cmd.Flags().XXXVarP`?
And if testing requires it, we could also move `NewDefaultCommandInitOption` to the corresponding test file.

Reply:
@tiansuo114  could you explain why we need it?

Reply:
> Could we consider removing the function `NewDefaultCommandInitOption`, as default values can all be set through `cmd.Flags().XXXVarP`?我们是否可以考虑删除函数 `NewDefaultCommandInitOption` ，因为默认值都可以通过 `cmd.Flags().XXXVarP` 设置？ And if testing requires it, we could also move `NewDefaultCommandInitOption` to the corresponding test file. 如果测试需要，我们也可以将 `NewDefaultCommandInitOption` 移动到相应的测试文件中。

I am attempting to discard the NewDefaultCommandInitOption method and use the following method to create CommandInitOption in the file.
```
func NewCmdConfigImagesList(parentCommand string) *cobra.Command {
	opts := configInit.CommandConfigImageOption{
		InitOption: &kubernetes.CommandInitOption{},
	}
	cmd := &cobra.Command{
```
However, when I actually execute it, the command produces the following error.
```
registry.k8s.io/kube-apiserver:
registry.k8s.io/kube-controller-manager:
registry.k8s.io/etcd:3.5.13-0



```
When the NewDefaultCommandInitOption initialization is not performed, the output string related to the image will be empty.
I haven't found a good way to solve this issue, so I have continued to keep the NewDefaultCommandInitOption method in use.

Reply:
So, the `karmadactl config image list` command currently does not support configuring image tags, correct?

Reply:
In what is now the code is to support the image of the tag, the image of the tags related to fill github.com/karmada-io/karmada/pkg/version.gitVersion fields at compile time

Reply:
@zhzhuang-zju 
I have added the output that the command would normally provide at the beginning of the PR. Regarding the question in the comment about the NewDefaultCommandInitOption method, what do you think would be the most appropriate way to handle this? Should we keep the currently working method, or do you have a better solution to suggest?

Reply:
> So, the command currently does not support configuring image tags, correct?karmadactl config image list

how about let's reduce review pressure to quickly move forward with PR consolidation, leave it to the next PR.

I think you're talking about the flag of `kube-image-tag` (we have not karmada-imag-tag yet)

Reply:
> I think you're talking about the flag of kube-image-tag (we have not karmada-imag-tag yet)

Yes, I think it would be much better to introduce the flag `kube-image-tag`. 
@tiansuo114  Could you help confirm whether the function `NewDefaultCommandInitOption` will still be needed after introducing the flag `kube-image-tag`? Thanks.

Reply:
@zhzhuang-zju 
In my understanding, the output of the config image list command consists of three parts:
repository link + image name + image tag
In this command, the first two parts rely on the initialization of kubernetes.CommandInitOption. Therefore, even after introducing kube-image-tag, the NewDefaultCommandInitOption method is still necessary.

Reply:
Thank you for your answer. If there are no current plans to expose more configurable flags, the current implementation is fine with me.

Reply:
get,I will fix the remaining issues as soon as possible and prepare for the next step